lib/active_model/serializer/adapter/json_api.rb

```diff
@@ -44,10 +46,17 @@ def fragment_cache(cached_hash, non_cached_hash)
 
         private
 
-        def add_relationships(resource, name, serializers)
+        def add_relationships(resource, name, serializers, opts)
           resource[:relationships] ||= {}
-          resource[:relationships][name] ||= { data: [] }
-          resource[:relationships][name][:data] += serializers.map { |serializer| { type: serializer.json_api_type, id: serializer.id.to_s } }
+
+          if opts.fetch(:data, true)
+            resource[:relationships][name] ||= { data: [] }
+            resource[:relationships][name][:data] += serializers.map { |serializer| { type: serializer.json_api_type, id: serializer.id.to_s } }
+          end
+          if opts.fetch(:links, false)
+            resource[:relationships][name] ||= {}
+            resource[:relationships][name][:links] = { related: url_helper.url_for([serializer.object, name]) }
```

Comment:
#1018 is kind stopped for sometime, I wouldn't worry abou that right now, but indeed:

> So the link building code should ideally be reusable in different contexts

This would be nice.